src/mlpack/methods/ann/layer_names.hpp

```diff
@@ -0,0 +1,316 @@
+/**
+ * @file layer_names.hpp
+ * @author Sreenik Seal
+ *
+ * Implementation of a class that converts a given ann layer to string format.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <boost/variant/static_visitor.hpp>
+#include <string>
+
+using namespace mlpack::ann;
+
+/**
+ * Implementation of a class that returns the string representation of the
+ * name of the given layer.
+ */
+class LayerNameVisitor : public boost::static_visitor<std::string>
+{
+ public:
+  //! Create the LayerNameVisitor object
+  LayerNameVisitor()
+  {
+  }
+
+  /*
+   * Return the name of the given layer of type AtrousConvolution as a string
```

Comment:
Missing stop at the end of the function and parameter descriptions.

Reply:
@zoq Could you take a look once? I looked into the test, it seems like it wasn't compiling because of some other issue (though still related to boost). I have pushed the test along with this but can you tell me how to run tests in general (like, what compilation statement do you use)?

Reply:
I'll add the new cpp file to the ``CMakeList.txt ``: https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/CMakeLists.txt right after ``lars_test.cpp`` should work just fine. Afterwards ``make`` should pick up the the file and include it into the build process.

If you make a test suite called "TestSuite" (``BOOST_AUTO_TEST_SUITE(TestSuite)``), and
then build ``mlpack_test`` (``make mlpack_test``) or just ``make``, you can run only the
tests in that test suite with ``bin/mlpack_test -t TestSuite``.  A
specific test case called ``TestCase (BOOST_AUTO_TEST_CASE(TestCase))``
could be run with ``bin/mlpack_test -t TestSuite/TestCase``.